api/pkg/s3/bucketlifecycleput.go

```diff
@@ -0,0 +1,135 @@
+	log.Logf("%x", md5.Sum(body))
+
+	if body != nil {
+		createLifecycleConf := model.LifecycleConfiguration{}
+		log.Logf("Before unmarshal struct is %v\n", createLifecycleConf)
+		err := xml.Unmarshal(body, &createLifecycleConf)
+		log.Logf("After unmarshal struct is %v\n", createLifecycleConf)
+		if err != nil {
+			response.WriteError(http.StatusInternalServerError, err)
+			return
+		} else {
+			s3RulePtrArr := make([]*s3.LifecycleRule, 0)
+			for _, rule := range createLifecycleConf.Rule {
+				s3Rule := s3.LifecycleRule{}
+
+				// assign the fields
+				s3Rule.ID = rule.ID //Assigning the rule ID
+
+				//Assigning the status value to s3 status
+				log.Logf("Status in rule file is %v\n", rule.Status)
+				s3Rule.Status = rule.Status
+
+				//Assigning the filter, using convert function to convert xml struct to s3 struct
+				s3Rule.Filter = convertRuleFilterToS3Filter(rule.Filter)
+
+				// Create the type of transition array
+				s3ActionArr := make([]*s3.Action, 0)
+
+				for _, transition := range rule.Transition {
+
+					//Defining the Transition array and assigning the values tp populate fields
+					s3Transition := s3.Action{Name: "Transition"}
+
+					//Assigning the value of days for transition to happen
+					s3Transition.Days = transition.Days
+
+					//Assigning the backend value to the s3 struct
+					s3Transition.Backend = transition.Backend
+
+					//Assigning the storage class of the object to s3 struct
+					//tier, err := s.class2tier(transition.StorageClass)
+					//if err != nil{
+					//	response.WriteError(http.StatusBadRequest, err)
+					//	return
+					//}
+					//s3Transition.Tier = tier
+
+					//Adding the transition value to the main rule
+					s3ActionArr = append(s3ActionArr, &s3Transition)
+				}
+
+				//Loop for getting the values from xml struct
+				for _, expiration := range rule.Expiration {
+					s3Expiration := s3.Action{Name: "Expiration"}
+					s3Expiration.Days = expiration.Days
+					s3ActionArr = append(s3ActionArr, &s3Expiration)
+				}
+				//Assigning the Expiration action to s3 struct expiration
+				s3Rule.Actions = s3ActionArr
+
+				s3Rule.AbortIncompleteMultipartUpload = convertRuleUploadToS3Upload(rule.AbortIncompleteMultipartUpload)
+				// add to the s3 array
+				s3RulePtrArr = append(s3RulePtrArr, &s3Rule)
+			}
+			// assign lifecycle rules to s3 bucket
+			bucket.LifecycleConfiguration = s3RulePtrArr
+			log.Logf("final bucket metadata is %v\n", bucket)
+		}
+	}
+
+	// Create bucket with bucket name will check if the bucket exists or not, if it exists
+	// it will internally call UpdateBucket function
+	res, err := s.s3Client.CreateBucket(ctx, bucket)
+	if err != nil {
+		response.WriteError(http.StatusInternalServerError, err)
+		return
+	}
+	log.Log("Create bucket lifecycle successfully.")
+	response.WriteEntity(res)
+
+}
+
+func convertRuleFilterToS3Filter(filter model.Filter) *s3.LifecycleFilter {
+	retFilter := s3.LifecycleFilter{}
+	retFilter.Prefix = filter.Prefix
```

Comment:
Need to check if filter is nil, otherwise there may panic.

Reply:
Done in the latest commit